shared/constants/transaction.js

```diff
@@ -0,0 +1,95 @@
+export const TRANSACTION_STATUS_APPROVED = 'approved'
+export const TRANSACTION_STATUS_CONFIRMED = 'confirmed'
+export const TRANSACTION_STATUS_UNAPPROVED = 'unapproved'
+export const TRANSACTION_STATUS_REJECTED = 'rejected'
+export const TRANSACTION_STATUS_SIGNED = 'signed'
+export const TRANSACTION_STATUS_SUBMITTED = 'submitted'
+export const TRANSACTION_STATUS_FAILED = 'failed'
+export const TRANSACTION_STATUS_DROPPED = 'dropped'
+
+// Transaction Category is a MetaMask construct used internally
+// Used to categorize transactions by purpose for displaying information to the user
+// 1. `'tranfser'`            - A token transaction where the user is sending tokens that they
+//                              own to another address
+// 2. `'transferfrom'`        - A token transaction transferring tokens from an account that
+//                              the sender has an allowance of (see the approve category).
+// 3. `'approve'`             - A token transaction requesting an allowance of the token to
+//                              spend on behalf of the user
+// 4. `'incoming'`            - An incoming (deposit) transaction
+// 5. `'sentEther'`           - A transaction sending ether to a recipient
+// 6. `'contractInteraction'` - A transaction that is interacting with a smart contract's methods
+// 7. `'contractDeployment'`  - A transaction that deployed a smart contract
+export const TRANSACTION_CATEGORY_TOKEN_METHOD_TRANSFER = 'transfer'
+export const TRANSACTION_CATEGORY_TOKEN_METHOD_APPROVE = 'approve'
+export const TRANSACTION_CATEGORY_TOKEN_METHOD_TRANSFER_FROM = 'transferfrom'
+export const TRANSACTION_CATEGORY_INCOMING = 'incoming'
+export const TRANSACTION_CATEGORY_SENT_ETHER = 'sentEther'
+export const TRANSACTION_CATEGORY_CONTRACT_INTERACTION = 'contractInteraction'
+export const TRANSACTION_CATEGORY_DEPLOY_CONTRACT = 'contractDeployment'
+
+// Transaction Group Status is a MetaMask construct to track status of groups
+// of transactions.
+// 1. `'cancelled'` - A cancel type transaction in the group was confirmed
+// 2. `'pending'`   - The primaryTransaction of the group has a status that is one of:
+//                    1. TRANSACTION_STATUS_APPROVED
+//                    2. TRANSACTION_STATUS_UNAPPROVED
+//                    3. TRANSACTION_STATUS_SUBMITTED
+export const TRANSACTION_GROUP_STATUS_CANCELLED = 'cancelled'
+export const TRANSACTION_GROUP_STATUS_PENDING = 'pending'
+
+// Transaction Group Category is a MetaMask construct to categorize the intent
+// of a group of transactions for purposes of displaying in the UI
+// 1. `'send'`              - Transaction group representing ether being sent from the user
+// 2. `'receive'`           - Transaction group representing a deposit/incoming transaction
+//                            usually maps 1:1 with TRANSACTION_CATEGORY_INCOMING
+// 3. `'interaction'`       - Transaction group representing an interaction with a smart
```

Comment:
It might be worth clarifying that these are just the interactions that we haven't special-cased. `approve` is a contract interaction as well.

Reply:
Was this comment addressed?